spec/lib/govuk_healthcheck/checkup_spec.rb

```diff
@@ -1,6 +1,64 @@
 require "spec_helper"
 require "govuk_app_config/govuk_healthcheck"
 
+class TestHealthcheck
+  def name
+    "#{status}_check".to_sym
+  end
+
+  def status
+    :unknown
+  end
+end
+
+class OkTestHealthcheck < TestHealthcheck
+  def status
+    :ok
+  end
+end
+
+class WarningTestHealthcheck < TestHealthcheck
+  def status
+    :warning
+  end
+end
+
+class CriticalTestHealthcheck < TestHealthcheck
+  def status
+    :critical
+  end
+end
+
+class OkTestHealthcheckWithMessage < OkTestHealthcheck
+  def message
+    "This is a custom message"
+  end
+end
+
+class OkTestHealthcheckWithDetails < OkTestHealthcheck
+  def details
+    {
+      extra: "This is an extra detail",
+    }
+  end
+end
+
+class DisabledCriticalHealthcheck < CriticalTestHealthcheck
+  def enabled?
+    false
+  end
+end
+
+class ExceptionHealthcheck
+  def name
+    :exception_check
+  end
+
+  def status
+    raise "something bad happened"
+  end
+end
```

Comment:
Rather than defining these classes in the global namespace I think we can create them with lets (since I think they only need to be accessed locally). For example:

```
  let(:test_healthcheck) do
    Class.new do
      def name
        "#{status}_check".to_sym
      end

      def status
        :unknown
      end
    end
  end

  let(:ok_check) do
    Class.new(test_healthcheck) do
      def status
        :ok
      end
    end
  end

  let(:warning_check) do
    Class.new(test_healthcheck) do
      def status
        :warning
      end
    end
  end

  let(:critical_check) do
    Class.new(test_healthcheck) do
      def status
        :critical
      end
    end
  end

  let(:disabled_critical_check) do
    Class.new(critical_check) do
      def enabled?
        false
      end
    end
  end

  let(:exception_check) do
    Class.new do
      def name
        :exception_check
      end

      def status
        raise "something bad happened"
      end
    end
  end
```

Reply:
Ah yep, I've fixed that in c81a6af.

Reply:
Nice one - just a heads up that you've still got the old classes defined.